### PR TITLE
docs: add http api

### PR DESCRIPTION
**Pull request by zhu-xiaowei**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary
1. add http api doc

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

File: docs/en/sdk-manual/index.md [hunk 1]
```diff
@@ -5,6 +5,8 @@
 - [Swift](./swift.md)  
 - [Web](./web.md)
 
+In addition, we also provide [HTTP API](./http-api.md) for directly report click stream data to the data pipeline.
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/index.md:8]: --> In addition, we also provide [HTTP API](./http-api.md) to collect clickstream data from other platforms (e.g., server) through http request.

File: docs/en/sdk-manual/http-api.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
+# Clickstream HTTP API
+
+## Introduction
+
+This documentation will help you send your data directly to the Clickstream ingestion server via HTTP requests, At the
+same time, following the below specifications, your data will be correctly processed by the Clickstream data processing
+module and can be visually analyzed in the subsequent report module.
```
Comment (luorobin-a2z) [inline docs/en/sdk-manual/http-api.md:7]: --> This article provides guidance for sending clickstream data to data pipeline through HTTP request. Following below specifications, the clickstream data will be processed properly by the data pipeline.

**Review by luorobin-a2z — COMMENTED** [review]
initial review

File: docs/en/sdk-manual/http-api.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+|--------------------------|-----------|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| appId                    | YES       | test_app            | The app id when creating the application can be obtained from the Clickstream Web Console                                                                  |
+| platform                 | NO        | Android/iOS/Web/... | Distinguish between platforms                                                                                                                              |
+| event_bundle_sequence_id | NO        | 1                   | Request sequence number, an incrementing integer starting from 1                                                                                           |
+| hashCode                 | NO        | 478acd09            | The first 8 digits of the sha256 calculation result of the request body string are used by the server to verify whether the request body has been modified |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:51]: -> The first eight digits of the sha256 calculation result of the request body string

File: docs/en/sdk-manual/http-api.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+| compression              | NO        | gzip                | Request body compression method. Currently only gzip is supported. Not passing this field means no compression                                             |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:52]: -> Request body compression method. Currently, only gzip is supported. Keeping it absent means no compression

File: docs/en/sdk-manual/http-api.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+### Request body
+
+The request body is an array structure, which contains the JSON string of one or more events. For example:
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:56]: -> The request body is an array structure that contains the JSON string of one or more events. For example:

File: docs/en/sdk-manual/http-api.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+| event_type       | YES      | String    | button_click                                                                                                                                                                                                                                                                                                                                     | Event name                                                                                                                                                                                                                                                                                                                                                                                                                                            |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:87]: `Event type` is better

File: docs/en/sdk-manual/http-api.md [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+| app_id           | YES      | String    | shopping_dev                                                                                                                                                                                                                                                                                                                                     | The corresponding id when creating the application in the Clickstream Web console                                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:90]: Web console -> web console

File: docs/en/sdk-manual/http-api.md [hunk 8]
```diff
@@ -0,0 +1,215 @@
...
+| unique_id        | NO       | String    | c84ad28d-16a8-4af4-a331-f34cdc7a7a18                                                                                                                                                                                                                                                                                                             | Unique ID to identify different users and associating the behavior of logged-in and not logged-in                                                                                                                                                                                                                                                                                                                                                     |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:93]: associating -> associate

File: docs/en/sdk-manual/http-api.md [hunk 9]
```diff
@@ -0,0 +1,215 @@
...
+| items            | NO       | Object    | [{<br/>	&nbsp;&nbsp;"id": "b011ddc3-632f-47cb-a68a-ad83678ecfed",<br/>	&nbsp;&nbsp;"name": "Classic coat-rack",<br/>	&nbsp;&nbsp;"category": "housewares",<br/>	&nbsp;&nbsp;"price": 167<br/>}]                                                                                                                                                  | Item list, Supports uploading multiple items at one time. A maximum of 100 items can be uploaded at one time<br>For the item quantity limit, please refer to [Event and Attribute Limitation](./web.md#event-and-attribute-limitation) <br>For the supported attributes of item, please refer to: [item attribute](./web.md#item-attributes)                                                                                                          |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:113]: -> For the supported attributes of the item, please refer to [item attribute](./web.md#item-attributes)

File: docs/en/sdk-manual/http-api.md [hunk 10]
```diff
@@ -0,0 +1,215 @@
...
+
+If the HttpCode status code returned by the request is 200, the request is considered successful, other status codes are failures, and the request does not return any other content.
+
+#### HttpCode
+
+| Code        | Message                        | Description                                                                |
+|-------------|--------------------------------|----------------------------------------------------------------------------|
+| 200         | OK                             | Request successful                                                         |
+| 413         | Request Entity Too Large       | Request failed. The request body exceeds 1MB                               |
+| 403         | DefaultAction: Invalid request | Request failed. Please check if appId and endpoint match                   |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:128]: remove message, it's not a public one

File: docs/en/sdk-manual/http-api.md [hunk 11]
```diff
@@ -0,0 +1,215 @@
...
+| 403         | Forbidden                      | Request failed. Please check whether the domain name resolution is correct |
```
Comment (zxkane) [inline docs/en/sdk-manual/http-api.md:129]: What's it? DNS resolution issue is out of the scope of API spec.

File: docs/en/sdk-manual/http-api.md (hunk 10 above)
Comment (zxkane) [reply docs/en/sdk-manual/http-api.md:128]: add `authentication` cookie checking

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/http-api.md [hunk 12]
```diff
@@ -0,0 +1,215 @@
...
+| Other Code  | --                             | Request failed                                                             |
+
+## Request code example
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:132]: changed to code blocks.

File: docs/mkdocs.en.yml [hunk 13]
```diff
@@ -53,6 +53,7 @@ nav:
       - Swift SDK: sdk-manual/swift.md
       - Web SDK: sdk-manual/web.md
       - Migrate from third-party SDKs: sdk-manual/migrate-from-third-party-sdks.md
+      - HTTP API: sdk-manual/http-api.md
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/mkdocs.en.yml:56]: changed the order.

File: docs/en/sdk-manual/http-api.md [hunk 14]
```diff
@@ -0,0 +1,215 @@
...
+This documentation will help you send your data directly to the Clickstream ingestion server via HTTP requests, At the
+same time, following the below specifications, your data will be correctly processed by the Clickstream data processing
+module and can be visually analyzed in the subsequent report module.
+
+## Request endpoint
+
+After creating the App in the Clickstream Web Console, you will get the request endpoint and appId on the App details
+page. For example:
+
+> https://example.com/collect
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:13]: changed the format and add **App ID**: `my_app`

File: docs/en/sdk-manual/http-api.md [hunk 15]
```diff
@@ -0,0 +1,215 @@
...
+   length of custom attribute value in attributes should not exceed 1024 characters.
+6. It is recommended that the number of events uploaded at a time should be controlled within 100.
+7. The total size of the body of a single request cannot exceed 1MB. If it exceeds, HttpCode 413 will be returned, error
+   content: `Request Entity Too Large`.
+
+### Reqeust method
+
+**`POST`**
+
+### Request header
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:35]: fixed

File: docs/en/sdk-manual/http-api.md [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+
+| Parameter name   | Required | Example                                                                                                                             | Description                                                                       |
+|------------------|----------|-------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|
+| Content-Type     | YES      | application/json; charset=utf-8                                                                                                     | Request Type                                                                      |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:39]: changed to content type

File: docs/en/sdk-manual/http-api.md [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+| X-Forwarded-For  | NO       | [101.188.67.134](https://whatismyipaddress.com/page/36)                                                                             | Source IP address                                                                 |
+| User-Agent       | NO       | Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36 | User-Agent，The browser will bring it by default when requesting                   |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:41]: removed.

File: docs/en/sdk-manual/http-api.md [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+### Reqeust method
+
+**`POST`**
+
+### Request header
+
+| Parameter name   | Required | Example                                                                                                                             | Description                                                                       |
+|------------------|----------|-------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------|
+| Content-Type     | YES      | application/json; charset=utf-8                                                                                                     | Request Type                                                                      |
+| X-Forwarded-For  | NO       | [101.188.67.134](https://whatismyipaddress.com/page/36)                                                                             | Source IP address                                                                 |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:40]: added.

File: docs/en/sdk-manual/http-api.md [hunk 19]
```diff
@@ -0,0 +1,215 @@
...
+| User-Agent       | NO       | Mozilla/5.0 (Linux; Android 6.0; Nexus 5 Build/MRA58N) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/116.0.0.0 Mobile Safari/537.36 | User-Agent，The browser will bring it by default when requesting                   |
+| cookie           | NO       | your auth cookie                                                                                                                    | Customize the verification cookie generated by OIDC for interface authentication  |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:42]: fixed.

File: docs/en/sdk-manual/http-api.md [hunk 20]
```diff
@@ -0,0 +1,215 @@
...
+
+### Request query parameters
+
+| Parameter name           | Required  | Example             | Description                                                                                                                                                |
+|--------------------------|-----------|---------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| appId                    | YES       | test_app            | The app id when creating the application can be obtained from the Clickstream Web Console                                                                  |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:48]: changed.

File: docs/en/sdk-manual/http-api.md (hunk 3 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:51]: changed to this.

File: docs/en/sdk-manual/http-api.md (hunk 4 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:52]: changed to this.

File: docs/en/sdk-manual/http-api.md (hunk 5 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:56]: changed to this.

File: docs/en/sdk-manual/http-api.md (hunk 6 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:87]: change to Event type.

File: docs/en/sdk-manual/http-api.md (hunk 7 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:90]: fixed

File: docs/en/sdk-manual/http-api.md (hunk 8 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:93]: fixed

File: docs/en/sdk-manual/http-api.md (hunk 10 above)
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:128]: fixed.

**Review by zxkane — CHANGES_REQUESTED** [review]

File: docs/en/sdk-manual/http-api.md [hunk 21]
```diff
@@ -0,0 +1,211 @@
...
+- **Server Endpoint**: `https://example.com/collect`
+- **App ID**: `my_app`
+
+## API Specification
+
+1. The app ID in the query parameters must be one of the applications you create for the project in the solution web
+   console. Otherwise, the server will respond to HTTP status code 403.
+2. The request body contains four parts: common attributes, items, user, and attributes. The public attributes require
+   the `event_type`, `event_id`, `timestamp`, and `app_id`; the rest are optional parameters.
+3. The total size of the body of a single request cannot exceed 1MB. The HTTP status code 403 will return if it exceeds.
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:24]: Yes, these copied from the above suggestion, fixed it.

File: docs/en/sdk-manual/http-api.md [hunk 22]
```diff
@@ -0,0 +1,211 @@
...
+failures, and the request does not return any other content.
+
+#### HttpCode
+
+| Code        | Description                                                                                                                                            |
+|-------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
+| 200         | Request successful                                                                                                                                     |
+| 403         | Request failed. Please check if appId and endpoint match, if configured with authentication, please check whether the authentication cookie is correct |
+| 413         | Request failed. The request body exceeds 1MB                                                                                                           |
+| Other Code  | Request failed                                                                                                                                         |
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/en/sdk-manual/http-api.md:124]: removed.

File: docs/zh/sdk-manual/http-api.md [hunk 23]
```diff
@@ -0,0 +1,205 @@
+# Clickstream HTTP API
+
+## 简介
+
+本文将帮助您通过 HTTP 请求将点击流数据发送到 Clickstream 提取服务器。 Clickstream
```
[the remark this answers is not in the document]
Comment (zhu-xiaowei) [reply docs/zh/sdk-manual/http-api.md:5]: fixed.

**Review by zxkane — APPROVED** [review]

**Pull request merged**